File: frontend/src/types/Job.ts
export interface Job {
  id: string;
  campaign_id: string;
  status: string;
  current_stage?: string;
  progress_percent: number;
  selected_slot_id: string | null;
  error_message: string | null;
  error_code: string | null;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
  metadata?: Record<string, unknown>;
}

export interface JobLog {
  timestamp: string;
  event_type: string;
  stage_name?: string;
  message: string;
}

File: backend/internal/models/job.go
package models

type Job struct {
	ID              string   `json:"id"`
	CampaignID      string   `json:"campaign_id"`
	Status          string   `json:"status"`
	CurrentStage    string   `json:"current_stage,omitempty"`
	ProgressPercent int      `json:"progress_percent"`
	SelectedSlotID  *string  `json:"selected_slot_id"`
	ErrorMessage    *string  `json:"error_message"`
	ErrorCode       *string  `json:"error_code"`
	CreatedAt       string   `json:"created_at"`
	StartedAt       *string  `json:"started_at"`
	CompletedAt     *string  `json:"completed_at"`
	Metadata        Metadata `json:"metadata,omitempty"`
}
